library/alloc/src/boxed.rs

```diff
@@ -1233,9 +1234,20 @@ impl<T: ?Sized, A: Allocator> Box<T, A> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_const_unstable(feature = "const_box", issue = "92521")]
 unsafe impl<#[may_dangle] T: ?Sized, A: Allocator> Drop for Box<T, A> {
+    #[inline]
     fn drop(&mut self) {
-        // FIXME: Do nothing, drop is currently performed by compiler.
+        // the T in the Box is dropped by the compiler before the destructor is run
+
+        let ptr = self.0;
+
+        unsafe {
+            let size = size_of_val(ptr.as_ref());
+            let align = min_align_of_val(ptr.as_ref());
+            let layout = Layout::from_size_align_unchecked(size, align);
```

Comment:
This could probably use `Layout::for_value` or even `Layout::for_value_raw`

Reply:
It could but that would likely regress compile time perf as more work has to be done. That's also why it uses intrinsics directly. Might be worth a try anyways.